src/main/java/org/logstash/snmp/SnmpClient.java

```diff
@@ -274,6 +278,28 @@ public <A extends Address> void processPdu(final CommandResponderEvent<A> event)
         }
     }
 
+    private boolean validateTrapMessage(int version, String securityName, int securityLevel, Set<String> allowedCommunities) {
+        if (version < SnmpConstants.version3) {
+            // Empty communities means "allow all communities" (BC with ruby-snmp)
+            if (!allowedCommunities.isEmpty() && !allowedCommunities.contains(securityName)) {
+                logger.debug("Received trap message with unknown community: '{}'. Skipping", securityName);
+                return false;
+            }
+        } else {
+            final Integer userSecurityLevel = this.usmUsersSecurityLevel.get(OctetString.fromString(securityName));
+            if (userSecurityLevel == null) {
```

Comment:
I think that if `security_name` is provided, `userSecurityLevel` _cannot_ be null because we inject the default `noAuthNoPriv` when `security_level` is unspecified.